landing-page

tjvsx opened the pull request.

tjvsx committed:
start

tjvsx committed:
responsive roadmap

tjvsx committed:
cleanup roadmap, update cover/banner, add blog & docs to navbar

tjvsx committed:
update social links

tjvsx committed:
update links

tjvsx committed:
scalable roadmap

tjvsx committed:
add aos, start lottie animation

tjvsx committed:
fix toggle's emoji size

tjvsx committed:
start lottie configs

tjvsx committed:
Update RoadmapSection.js

tjvsx committed:
play animations subsequently

tjvsx committed:
animation completed

mwaa commented:
@tjvsx Looking into the code but currently concerned about the load speed of the landing page. Resources are heavy, lighthouse is unable to process page

tjvsx commented:
@mwaa I agree, I think it's time to focus on performance. My first thoughts are
- reduce size of png banner/cover photo
- use lazy loading
- simplify/lighten the code
- remove animation's 'complete' even listener after use
... I will research some additional ways, but would love to hear your thoughts and ideas on this

tjvsx committed:
start optimization

tjvsx committed:
clean up

tjvsx committed:
optimization update

- Used imagekit fetching for png -> webp format
- Use CSS imports on each section
- Split the cover/banner and the landing page, and lazy loaded the landing page as a bundle

tjvsx committed:
fixed margins, sizing, diamond mobile img

tjvsx committed:
fix diamond-image

tjvsx committed:
shapes' preserveaspectratio switch

mwaa commented:
> @mwaa I agree, I think it's time to focus on performance. My first thoughts are
> 
> * reduce size of png banner/cover photo
> * use lazy loading
> * simplify/lighten the code
> * remove animation's 'complete' even listener after use
>   ... I will research some additional ways, but would love to hear your thoughts and ideas on this

This should be sufficient i guess. 
If images are lazy loaded then should be better

mwaa commented on docusaurus.config.js:
```diff
@@ -29,17 +29,34 @@ const config = {
   //     },
   //   ],
   // ],
+  // plugins: [
```
If you evaluate the plugin is not needed then rather than comment more code it would be preferable to remove entirely

mwaa commented on package.json:
```diff
@@ -16,14 +16,18 @@
   "dependencies": {
     "@cmfcmf/docusaurus-search-local": "^0.10.0",
```
In the config this is commented out, if not needed also remove the dependency

mwaa commented on sidebars.js:
```diff
@@ -17,15 +17,15 @@ const sidebars = {
-  ],
-   */
+  
+  // indexSidebar: [
```
Remove code instead of comments if not needed

mwaa commented on src/css/custom.css:
```diff
@@ -14,7 +14,26 @@
+  --ifm-color-primary-lighter: #948fff;
+  --ifm-color-primary-lightest: #c6c3ff;
+}
+/* html[data-theme='dark'] {
```
Same clean up all code comments if not needed or move to reference developer doc if you need to maintain color reference in future

mwaa commented on src/css/custom.css:
```diff
@@ -162,53 +160,72 @@ div[class*="toggleTrack_"] {
 
 div[class*="toggleTrackThumb"] {
-  background: url('/img/emoji-sunrise-over-mountains.png') no-repeat;
+  background: url('https://ik.imagekit.io/habitat/emoji-sunrise-over-mountains.png?ik-sdk-version=javascript-1.4.3&updatedAt=1645306385939') no-repeat;
```
Nice use of CDN. 
Are the images to the CDN compressed for web?
Not sure if you have still retained a local copy of the images folder and add in that folder a README.md file that the images have been uploaded to imagekit CDN to assist future developers when they want to change an image

mwaa commented on src/pages/sections/InfoSection.js:
```diff
@@ -0,0 +1,46 @@
+      <div id='info_wrapper'>
+        <div id='info_container'>
+          <div className='info_feature'>
+            <Optimism className={clsx('info_img', styles.svg_img)} height={100} width={100}/>
```
Quick question what do you think of className={`info_img ${styles.svg_img}`}?
Instead of using clsx and remove it as a dependency in an effort to reduce dependencies?

mwaa replied:
This would apply on each usage of clsx.
Just a question need you opinion if you feel its worth using clsx

tjvsx replied:
this was what docusaurus was using on their template landing page. there may be an underlying reason for its use.

mwaa replied:
Cool wee can keep for now

mwaa reviewed (changes requested):
Good work @tjvsx 
Minor changes but otherwise performance has improved considerably. 

- Can you change the blog to point to substack for now 
- Can you add SEO optimization https://docusaurus.io/docs/next/seo  basically page meta for layout page is only thing missing

tjvsx committed:
optimize for desktop, add images folder w/ README

tjvsx committed:
style detailing

mwaa commented on docusaurus.config.js:
```diff
@@ -6,40 +6,33 @@ const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+  title: 'Habitat',
   tagline: 'Start your organization on Optimism',
-  url: 'https://github.com/0xHabitat/docs',
+  url: 'https://habitat/',
```
Tiny missed url should be https://0xhabitat.org/

mwaa commented on src/pages/index.js:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <Layout
+      title='Home'
+      description="Description will go into a meta tag in <head />">
```
Can we change this description or its overloaded by something in the config?

tjvsx committed:
minor fixes

tjvsx committed:
token section CTA
